cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java

```diff
@@ -89,7 +100,17 @@ public ApplicationManager deployApplication(Id.Namespace namespace,
     // Should never happen, otherwise the ClassLoader is broken
     Preconditions.checkNotNull(appClassURL, "Cannot find class %s from the classloader", applicationClz);
 
+    String appConfig = "";
+    TypeToken typeToken = TypeToken.of(applicationClz);
+    TypeToken<?> configToken = typeToken.resolveType(Application.class.getTypeParameters()[0]);
+
     try {
+      if (configObject != null) {
+        appConfig = GSON.toJson(configObject, configToken.getType());
```

Comment:
i think you should be able to just pass the config object without the type. if not, can you add a comment explaining why it needs to be resolved.